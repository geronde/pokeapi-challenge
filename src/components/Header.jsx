import React from 'react'
import styled from 'styled-components';

const Wrapper = styled.header`
grid-area: header;
height: 180px;
padding: 20px;
font-size: 30px;
font-weight: bold;
margin-bottom: 20px;
background-color: #0E1111;
color: #eee;
display: flex;
align-items: center;
justify-content: center;
`

const Header = (props) => {
    return <Wrapper>{props.children}</Wrapper>

}

export default Header;