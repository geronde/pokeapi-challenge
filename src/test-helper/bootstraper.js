import faker from 'faker';

export const generateTestPokemons = (n = 15, isType) => {
  const pokemons = [];

  for (let i = 0; i < n; i++) {
    let poke = { name: faker.random.word(), url: `fake-url/${i}/` };
    if (isType) {
      poke = { pokemon: { ...poke } };
    }
    pokemons.push(poke);
  }
  return pokemons;
};
