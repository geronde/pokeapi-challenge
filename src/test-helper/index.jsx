import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

export const render = (ui, { ...renderOptions } = {}) =>{
    function Wrapper({ children }) {
      return <BrowserRouter>{children}</BrowserRouter>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  }