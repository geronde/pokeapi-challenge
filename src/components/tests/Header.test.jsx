import React from "react";
import {render} from '../../test-helper/index'
import Header from '../Header'

describe('Header', () => {
    let props;
    beforeEach(() => {
        props = {
            children: <div>header</div>
        }
    });
    it('should display header correctly', () => {
       const { asFragment } = render(<Header {...props} />)
       expect(asFragment()).toMatchSnapshot();
    });
});