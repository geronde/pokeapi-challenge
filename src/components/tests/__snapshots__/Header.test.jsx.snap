// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should display header correctly 1`] = `
<DocumentFragment>
  .c0 {
  grid-area: header;
  height: 180px;
  padding: 20px;
  font-size: 30px;
  font-weight: bold;
  margin-bottom: 20px;
  background-color: #0E1111;
  color: #eee;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<header
    class="c0"
  >
    <div>
      header
    </div>
  </header>
</DocumentFragment>
`;
